autofaiss/external/quantize.py

```diff
@@ -290,7 +290,7 @@ def write_ids_df_to_parquet(ids: pd.DataFrame, batch_id: int):
         if nb_indices_to_keep > 1:
             indices_folder = cast(str, indices_folder)
             max_nb_threads = max(
-                1,
```

Comment:
this make max_nb_thread have a minimum value of multiprocessing.cpu_count()
max_nb_thread >= multiprocessing.cpu_count()

I think what we want is the opposite : max_nb_thread <= multiprocessing.cpu_count()
so that mean min(multiprocessing.cpu_count(), max(1, math.floor(
                    cast_memory_to_bytes(current_memory_available)
                    / (cast_memory_to_bytes(max_index_memory_usage) / nb_indices_to_keep) ))

(keeping the 1 because 0 is not a good value)

Reply:
Indeed, thanks for pointing it out. It was a silly mistake !